src/app/pages/storage2/modules/devices/components/zfs-info-card/zfs-info-card.component.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  ChangeDetectionStrategy, Component, Input, OnInit,
+} from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute } from '@angular/router';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { filter } from 'rxjs/operators';
+import { VDevType } from 'app/enums/v-dev-type.enum';
+import { VDevStatus } from 'app/enums/vdev-status.enum';
+import { Disk, VDev } from 'app/interfaces/storage.interface';
+import { EntityJobComponent } from 'app/modules/entity/entity-job/entity-job.component';
+import { EntityUtils } from 'app/modules/entity/utils';
+import { DiskFormComponent } from 'app/pages/storage/disks/disk-form/disk-form.component';
+import { DevicesStore } from 'app/pages/storage2/modules/devices/stores/devices-store.service';
+import { WebSocketService, DialogService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-zfs-info-card',
+  templateUrl: './zfs-info-card.component.html',
+  styleUrls: ['./zfs-info-card.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ZfsInfoCardComponent implements OnInit {
+  @Input() topologyItem: VDev;
+  @Input() topologyParentItem: VDev;
+  @Input() disk: Disk;
+  private poolId: number;
+
+  get readErrors(): number {
+    if (this.isMirror) {
+      return this.topologyItem.children.reduce((errors, vdev) => {
+        return errors + (vdev.stats?.read_errors || 0);
+      }, 0);
+    }
+    return this.topologyItem.stats.read_errors;
+  }
+
+  get writeErrors(): number {
+    if (this.isMirror) {
+      return this.topologyItem.children.reduce((errors, vdev) => {
+        return errors + (vdev.stats?.write_errors || 0);
+      }, 0);
+    }
+    return this.topologyItem.stats.write_errors;
+  }
+
+  get checksumErrors(): number {
+    if (this.isMirror) {
+      return this.topologyItem.children.reduce((errors, vdev) => {
+        return errors + (vdev.stats?.checksum_errors || 0);
+      }, 0);
+    }
+    return this.topologyItem.stats.checksum_errors;
+  }
+
+  get isMirror(): boolean {
+    return this.topologyItem.type === VDevType.Mirror;
+  }
+
+  get isDisk(): boolean {
+    return this.topologyItem.type === VDevType.Disk;
+  }
+
+  get isOnline(): boolean {
+    return this.topologyItem.status === VDevStatus.Online;
+  }
+
+  constructor(
+    private route: ActivatedRoute,
+    private ws: WebSocketService,
+    private slideInService: IxSlideInService,
+    private dialogService: DialogService,
+    private matDialog: MatDialog,
+    private translate: TranslateService,
+    private devicesStore: DevicesStore,
+  ) {}
+
+  ngOnInit(): void {
+    this.poolId = +this.route.snapshot.paramMap.get('poolId');
+    this.slideInService.onClose$?.pipe(untilDestroyed(this)).subscribe(() => {
```

Comment:
This will likely always reload disk even if the model is closed by clicking cancel/cross button. Let's make sure all instances of `sliceInService.close()` in the form return `true` for successful changes and then check here for that boolean. So we can only reload when required and not always.

Reply:
Sounds reasonable but it seems there are no places where it is implemented. I'll keep consistency with DiskInfoWidget and other forms for now.